### PR TITLE
Remove boost dependency

### PR DESCRIPTION
**Pull request by jerinphilip**
Necessary requirements to change L4 to work without boost. 

1. `boost::interprocess:vector` -> `std::vector`
2. `boost::interprocess::detail::*` -> pulled some sources/functions from boost source. Might want to get rid of these. 

The non-process stuff should work accordingly.

### CODE REVIEW

File: inc/L4/Epoch/EpochQueue.h [hunk 1]
```diff
@@ -38,8 +38,10 @@ struct EpochQueue {
   using SharableLock = TSharableLock;
   using ExclusiveLock = TExclusiveLock;
   using RefCount = std::atomic<std::uint32_t>;
-  using RefCounts = Interprocess::Container::
-      Vector<RefCount, typename Allocator::template rebind<RefCount>::other>;
+  // @jerinphilip: dirty-fix, browsermt/L4 hopefully does not need IPC.
```
Comment (XapaJIaMnu) [inline inc/L4/Epoch/EpochQueue.h:41]: What is IPC? Replacing interprocess::vector with vector will lead to a race condition. THe questions is, do we care that we have a race condition.

File: inc/L4/detail/ToRawPointer.h [hunk 2]
```diff
@@ -1,11 +1,139 @@
 #pragma once
 
-#include <boost/interprocess/detail/utilities.hpp>
 
 namespace L4 {
 namespace Detail {
 
-using boost::interprocess::ipcdetail::to_raw_pointer;
+//#include <boost/interprocess/detail/utilities.hpp>
```
Comment (XapaJIaMnu) [inline inc/L4/detail/ToRawPointer.h]: Is this a copy from the Boost codebase?

**Review by XapaJIaMnu — APPROVED** [review]
Could you add some more details on what exactly  we lose with the loss of boost::interprocess? Is it used for the LRU(ish) eviction policy?

File: inc/L4/detail/ToRawPointer.h (hunk 2 above)
Comment (jerinphilip) [reply inc/L4/detail/ToRawPointer.h]: Yes.

Comment (XapaJIaMnu) [reply inc/L4/detail/ToRawPointer.h]: Say so

File: inc/L4/Epoch/EpochQueue.h (hunk 1 above)
Comment (jerinphilip) [reply inc/L4/Epoch/EpochQueue.h:41]: I wrote it short for inter-process-communication. @kpu told me at some point it's okay and just offsets, where offsets become 0 or so when not in a multiproc setting (our use-case). I haven't been able to find much documentation on this.

**Comment by jerinphilip** [comment]
> Could you add some more details on what exactly we lose with the loss of boost::interprocess? 

TBH, I don't know. It'd take me a while to figure starting from zero multiproc exp. I figured throw a lot of sentences, cached counterpart to cache and report if things fail, it didn't fail.

> Is it used for the LRU(ish) eviction policy?

LRU what? I've checked in a lazy LRU for WebAssembly. L4 follows the Clock eviction policy. This is an approximation for LRU, where you remove things either if they have been accessed once or you run out of memory. The Eviction policy is independent of IPC (here through shared-memory?).

File: inc/L4/Epoch/EpochQueue.h (hunk 1 above)
Comment (XapaJIaMnu) [reply inc/L4/Epoch/EpochQueue.h:41]: i guess it's fine then. Tests pass, it's not obviously broken? xD

**Pull request merged**